src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponentDescriptorCollection.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    /// <summary>
+    /// A cached collection of <see cref="ViewComponentDescriptor" />.
+    /// </summary>
+    public class ViewComponentDescriptorCollection
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ViewComponentDescriptorCollection"/>.
+        /// </summary>
+        /// <param name="items">The result of view component discovery</param>
+        /// <param name="version">The unique version of discovered view components.</param>
+        public ViewComponentDescriptorCollection([NotNull] IReadOnlyList<ViewComponentDescriptor> items, int version)
+        {
+            Items = items;
```

Comment:
should this constructor accept an `IEnumerable<ViewComponentDescriptor>` and create its own `IReadOnlyList<ViewComponentDescriptor>` to avoid issues if the caller (messes up and) changes their underlying collection?